src/ui/public/vis/vis_types/react_vis_type.js

```diff
@@ -19,49 +19,44 @@
 
 import React from 'react';
 import { render, unmountComponentAtNode } from 'react-dom';
-import { VisTypeProvider } from './';
+import chrome from '../../chrome';
+import { VisType } from './base_vis_type';
 
-export function ReactVisTypeProvider(Private, getAppState, config) {
-  const VisType = Private(VisTypeProvider);
-
-  class ReactVisController {
-    constructor(el, vis) {
-      this.el = el;
-      this.vis = vis;
-    }
-
-    render(visData, updateStatus) {
-      this.visData = visData;
+class ReactVisController {
+  constructor(el, vis) {
+    this.el = el;
+    this.vis = vis;
+  }
 
-      return new Promise((resolve) => {
-        const Component = this.vis.type.visConfig.component;
-        render(<Component
-          config={config}
-          vis={this.vis}
-          appState={getAppState()}
-          visData={visData}
-          renderComplete={resolve}
-          updateStatus={updateStatus}
-        />, this.el);
-      });
-    }
+  render(visData, updateStatus) {
+    this.visData = visData;
+
+    return new Promise((resolve) => {
+      const Component = this.vis.type.visConfig.component;
+      const config = chrome.getUiSettingsClient();
+      render(<Component
+        config={config}
+        vis={this.vis}
+        visData={visData}
+        renderComplete={resolve}
+        updateStatus={updateStatus}
+      />, this.el);
+    });
+  }
 
-    destroy() {
-      unmountComponentAtNode(this.el);
-    }
+  destroy() {
+    unmountComponentAtNode(this.el);
   }
+}
 
-  class ReactVisType extends VisType {
-    constructor(opts) {
-      opts.visualization = ReactVisController;
+export class ReactVisType extends VisType {
+  constructor(opts) {
+    opts.visualization = ReactVisController;
 
-      super(opts);
+    super(opts);
```

Comment:
what if we move the super call as first statement and than, having the `this` available, set the opts.visualization to ReactVisController?
If not, this seems more a Factory method than a constructor.

Reply:
base vis type will complain about visualization controller missing ... but i joined the two lines in one to make it more readable